### PR TITLE
Avoid Dynamo crash when running the migration tool on clockwork 1.3 package. 

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

This PR address the crash issue that was happening when using the migration tool on clockwork 1.3 package. 
https://jira.autodesk.com/browse/QNTM-5616

Along with this change, some changes have to be made to the instructions displayed on the migration tool window. I will be creating another PR for that. 

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@alfarok @QilongTang  @mjkkirschner 

### FYIs

@ColinDayOrg

### CODE REVIEW

File: src/LibraryViewExtension/Handlers/NodeItemDataProvider.cs [hunk 1]
```diff
@@ -74,9 +74,9 @@ protected Stream GetNodeItemDataStream(IEnumerable<NodeSearchElement> searchEntr
         protected virtual object CreateObjectForSerialization(IEnumerable<NodeSearchElement> searchEntries)
         {
             var data = new LoadedTypeData<LoadedTypeItem>();
-            data.loadedTypes = searchEntries
-                //.Where(e => !e.ElementType.HasFlag(ElementTypes.Packaged))
-                .Select(e => CreateLoadedTypeItem<LoadedTypeItem>(e)).ToList();
+
+            // Converting searchEntries to another list so as to avoid modifying the actual searchEntries list when iterating through it.
```
Comment (mjkkirschner) [inline src/LibraryViewExtension/Handlers/NodeItemDataProvider.cs]: why was this condition removed?
`!e.ElementType.HasFlag(ElementTypes.Packaged))`

**Comment by mjkkirschner** [comment]
@reddyashish what is the migration tool window?

File: src/LibraryViewExtension/Handlers/NodeItemDataProvider.cs (hunk 1 above)
Comment (QilongTang) [reply src/LibraryViewExtension/Handlers/NodeItemDataProvider.cs]: Was it used before, it was commented out and not sure for what reason

Comment (mjkkirschner) [reply src/LibraryViewExtension/Handlers/NodeItemDataProvider.cs]: ahh sorry, overlooked that. 😢

**Comment by mjkkirschner** [comment]
While this fix gets around the issue, I may be thinking about it incorrectly, but does it not also imply that we have an underlying concurrency issue? I am wondering if a better long term solution would be a lock or task structure.

**Comment by mjkkirschner** [comment]
@reddyashish Ahhh! I get it now, you used the migration extension to reproduce this bug!

**Comment by QilongTang** [comment]
@mjkkirschner Yes, so I suggested we put some limitation info on Migration Extension UI because we know that it would not be able to handle 200 files in a row 😉

**Comment by mjkkirschner** [comment]
@QilongTang because of this issue or because of a memory leak?

**Comment by QilongTang** [comment]
@mjkkirschner After this fix. Memory leak still happening which is a blocker for giant folder migration.

**Comment by reddyashish** [comment]
Sorry I was out for sometime to collect my package. @mjkkirschner yes, the crash was reproduced with the migration tool on clockwork package. I did get any crash with only installing a bunch of packages. I verified the exception again to be sure of the crash that was reported. 

And there is memory leak when we run the tool on more than 200 packages which sometimes crashes the revit+dynamo and sometimes goes into not responding state. 

I was going to create another issue for the memory leak and then take it from there. What do you think?

**Comment by alfarok** [comment]
@reddyashish That sounds good to me.
- Make a note on the upgrade tool view extension repo possibly recommending updates 50-100 files at a time max
- File any notes you have on the memory leak in a new task

**Comment by ColinDayOrg** [comment]
LGTM

**Comment by mjkkirschner** [comment]
I'd like to point out that the migration extension not working on large packages is kind of annoying but in my opinion not as important as getting to the bottom of this issue - I'm curious if you discovered why the searchElements are being modified while they are being iterated?

**Pull request merged**